src/binary-reader-objdump.c

```diff
@@ -271,18 +272,22 @@ static WasmResult on_opcode_uint64(WasmBinaryReaderContext* ctx,
 }
 
 static WasmResult on_opcode_f32(WasmBinaryReaderContext* ctx,
-                                float value) {
+                                uint32_t value) {
   Context* context = ctx->user_data;
   size_t immediate_len = ctx->offset - context->current_opcode_offset;
-  log_opcode(context, ctx->data, immediate_len, "%g", value);
+  char buffer[20];
+  wasm_write_float_hex(buffer, sizeof(buffer), value);
+  log_opcode(context, ctx->data, immediate_len, buffer);
   return WASM_OK;
 }
 
 static WasmResult on_opcode_f64(WasmBinaryReaderContext* ctx,
-                                double value) {
+                                uint64_t value) {
   Context* context = ctx->user_data;
   size_t immediate_len = ctx->offset - context->current_opcode_offset;
-  log_opcode(context, ctx->data, immediate_len, "%g", value);
+  char buffer[20];
```

Comment:
looks like this can actually be 25 characters, including null terminator. (see comment at the top of wasm_write_double_hex). I thought I added a #define for this in literal.h, but I don't see it. It's probably worth adding one, if you don't mind.